src/linear_algebra/matrix.lean

```diff
@@ -936,3 +936,43 @@ def alg_equiv_matrix {R : Type v} {M : Type w} {n : Type*} [fintype n]
   [comm_ring R] [add_comm_group M] [module R M] [decidable_eq n] {b : n → M} (h : is_basis R b) :
   module.End R M ≃ₐ[R] matrix n n R :=
 h.equiv_fun.alg_conj.trans alg_equiv_matrix'
+
+/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
+def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
+  (i : n) : n → R := linear_map.std_basis R (λ _, R) i 1
+
+namespace std_basis_vec
+
+open_locale classical
+
+variables {R : Type v} [ring R] {n : Type w}
+
+lemma neq_eq_zero {i j : n} (h : i ≠ j) : std_basis_vec i j = (0 : R) :=
+by { rw [std_basis_vec, linear_map.std_basis_ne], exact h.symm }
+
+lemma eq_eq_one {i j : n} (h : i = j) : std_basis_vec i j = (1 : R) :=
+by { convert linear_map.std_basis_same _ (λ _, R) j 1 }
+
+lemma is_basis [fintype n] : @is_basis n R (n → R) std_basis_vec _ _ _ :=
+  pi.is_basis_fun R n
+
+lemma dot_product_eq_val (v : n → R) (i : n) [fintype n]:
+  v i = matrix.dot_product v (std_basis_vec i) :=
+begin
+  rw [matrix.dot_product, finset.sum_eq_single i, eq_eq_one rfl, mul_one],
+  exact λ _ _ hb, by rw [neq_eq_zero hb.symm, mul_zero],
+  exact λ hi, false.elim (hi $ finset.mem_univ _)
+end
+
+end std_basis_vec
+
+lemma matrix.dot_product_eq_zero {R : Type v} [ring R] {n : Type w} [fintype n]
```

Comment:
Two comments:

1. I think it is confusing that `v` is both a variable name and a universe. You can write `Type*` instead of `Type v`, and lean will do the right thing. Same for `w`.
2. This lemma seems a special case of a more general lemma: two vectors are equal if their dot products with all `w` are equal.

Reply:
Is the more general lemma provable as `matrix.to_lin'.injective $ linear_map.ext h` or similar?